src/Learners/Recommender/MatchboxRecommender.cs

```diff
@@ -139,17 +141,18 @@ public static class MatchboxRecommender
         /// <typeparam name="TFeatureSource">The type of a feature source.</typeparam>
         /// <param name="stream">The binary stream to a serialized Matchbox recommender.</param>
         /// <param name="mapping">The mapping used for accessing data in the native format.</param>
+        /// <param name="formatter">The formatter to use.</param>
         /// <returns>The deserialized recommender object.</returns>
         public static IMatchboxRecommender<TInstanceSource, int, int, Discrete, TFeatureSource>
             LoadBackwardCompatible<TInstanceSource, TFeatureSource>(
-                Stream stream, IMatchboxRecommenderMapping<TInstanceSource, TFeatureSource> mapping)
+                Stream stream, IMatchboxRecommenderMapping<TInstanceSource, TFeatureSource> mapping, IFormatter formatter)
         {
             if (stream == null)
             {
                 throw new ArgumentNullException(nameof(stream));
             }
 
-            using (var reader = new WrappedBinaryReader(new BinaryReader(stream, Encoding.UTF8, true)))
+            using (var reader = new WrappedBinaryReader(new BinaryReader(stream, Encoding.UTF8, true), formatter))
```

Comment:
We don't always use Json internally.

But since the TextReader overloads work, it seems like we never refer through to the WrappedBinaryReader's IFormatter. So yes, I could change it to be null and remove these in this case.

Reply:
Give me an example.